config/initializers/intercom.rb

```diff
@@ -0,0 +1,118 @@
+if ENV["INTERCOM_APP_ID"].present? && ENV["INTERCOM_IDENTITY_VERIFICATION_KEY"].present?
+  IntercomRails.config do |config|
+    # == Intercom app_id
+    #
+    config.app_id = ENV["INTERCOM_APP_ID"]
+
+    # == Intercom session_duration
+    #
+    # config.session_duration = 300000
+    # == Intercom secret key
+    # This is required to enable Identity Verification, you can find it on your Setup
+    # guide in the "Identity Verification" step.
+    #
+    config.api_secret = ENV["INTERCOM_IDENTITY_VERIFICATION_KEY"]
+
+    # == Enabled Environments
+    # Which environments is auto inclusion of the Javascript enabled for
+    #
+    config.enabled_environments = [ "development", "production" ]
+
+    # == Current user method/variable
+    # The method/variable that contains the logged in user in your controllers.
+    # If it is `current_user` or `@user`, then you can ignore this
+    #
+    config.user.current = Proc.new { Current.user }
+
+    # == Include for logged out Users
+    # If set to true, include the Intercom messenger on all pages, regardless of whether
+    # The user model class (set below) is present.
+    config.include_for_logged_out_users = true
+
+    # == User model class
+    # The class which defines your user model
+    #
+    # config.user.model = Proc.new { User }
+
+    # == Lead/custom attributes for non-signed up users
+    # Pass additional attributes to for potential leads or
+    # non-signed up users as an an array.
+    # Any attribute contained in config.user.lead_attributes can be used
+    # as custom attribute in the application.
+    # config.user.lead_attributes = %w(ref_data utm_source)
+
+    # == Exclude users
+    # A Proc that given a user returns true if the user should be excluded
+    # from imports and Javascript inclusion, false otherwise.
+    #
+    # config.user.exclude_if = Proc.new { |user| user.deleted? }
+
+    # == User Custom Data
+    # A hash of additional data you wish to send about your users.
+    # You can provide either a method name which will be sent to the current
+    # user object, or a Proc which will be passed the current user.
+    #
+    config.user.custom_data = {
+      family_id: Proc.new { |current_user| current_user.family.id }
+    }
+
+    # == Current company method/variable
+    # The method/variable that contains the current company for the current user,
+    # in your controllers. 'Companies' are generic groupings of users, so this
+    # could be a company, app or group.
+    #
+    # config.company.current = Proc.new { current_company }
```

Comment:
Not entirely sure how this shows up in Intercom UI, but could potentially be repurposed?

```rb
config.company.current = Proc.new { Current.family }
```

Reply:
Oooo, interesting idea. Let me see how that filters down through Intercom.

Reply:
Done. Super easy.